strawberry/enum.py

```diff
@@ -11,16 +11,27 @@ class EnumValue:
     value: Any
 
 
-@dataclasses.dataclass
-class EnumDefinition:
+@dataclasses.dataclass(frozen=True)
+class StrawberryEnum:
+    enum: EnumMeta
+    # TODO: graphql_name, python_name
     name: str
-    values: List[EnumValue]
+    values: List[EnumValue] = dataclasses.field(hash=False)
     description: Optional[str]
 
+    def __call__(self, *args: Any, **kwds: Any) -> Any:
+        return self.enum(*args, **kwds)
+
+    def __getattr__(self, attr):
+        if hasattr(self.enum, attr):
+            return self.enum[attr]
+
+        return super().__getattribute__(attr)
```

Comment:
I need a sanity check on this :D

Reply:
I think this is correct (`object` doesn't define `__getattr__` by default, only `__getattribute__`), but I would type annotate this function a bit. This should always take a `str` and return an `Enum`, right?

Reply:
yes! the return type depends on what attribute we are requesting though